### PR TITLE
Signup Page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Signup Page.

- This PR has components and the APIs required for signup page.

- Files to review:
   - [ ] src/pages/Signup.js
   - [ ] src/components/Signup/SignupForm.js
   - [ ] src/contexts/signup.context.js
   - [ ] src/contexts/auth.context.js
   - [ ] src/reducers/signup.reducer.js

v-1

- Use this link for preview - https://deploy-preview-1--pensive-hodgkin-f2cf81.netlify.app

Thanks.

### CODE REVIEW

**Review by Maianki — APPROVED** [review]
Nice work Vinay. Nothing to suggest. 
Your code is very well structured and readable. Good to merge.

File: src/reducers/signup.reducer.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            ...state, confirmedPassword: {
+                ...state.confirmedPassword,
+                error: true
+            }
+        }
+        case 'REMOVE_CONFIRMED_PASSWORD_ERROR': return {
+            ...state, confirmedPassword: {
+                ...state.confirmedPassword,
+                error: false
+            }
```
Comment (samyukthasudhakar) [inline src/reducers/signup.reducer.js:119]: Appreciation : Like how you've handled all types error in here.

**Review by samyukthasudhakar — APPROVED** [review]
The code looks great - clean and well structured  ! I do not have any suggestions. This is good to merge.

File: src/reducers/signup.reducer.js (hunk 1 above)
Comment (foolhardy21) [reply src/reducers/signup.reducer.js:119]: thankyou

**Pull request merged**